digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/LinearAssetService.scala

```diff
@@ -794,7 +794,8 @@ trait LinearAssetOperations {
   }
 
   def adjustedSideCode(adjustment: SideCodeAdjustment): Unit = {
-    val oldAsset = getPersistedAssetsByIds(adjustment.typeId, Set(adjustment.assetId), newTransaction = false).head
+    val oldAsset = getPersistedAssetsByIds(adjustment.typeId, Set(adjustment.assetId), newTransaction = false).headOption
+      .getOrElse(throw new IllegalStateException("Old asset " + adjustment.assetId + " no longer available"))
```

Comment:
Tää on tälläinen geneerinen, joten ehkä tyyppi tiedon antaminen myös vois olla hyödyllinen.

Reply:
.getOrElse(throw new IllegalStateException("Old asset " + adjustment.assetId + **" of type " + adjustment.typeId** + " no longer available")) oisko tämmönen parempi siihen?